openfasoc/MLoptimization/model.py

```diff
@@ -0,0 +1,47 @@
+#training import
+import gym
+import ray
+import ray.tune as tune
+from ray.rllib.algorithms.ppo import PPO
+from run_training import Envir
+from sky130_nist_tapeout import single_build_and_simulation
+sky130_nist_tapeout.path.append('../generators/gdsfactory-gen/')
+
+import argparse
+#
+#training set up
+parser = argparse.ArgumentParser()
+parser.add_argument('--checkpoint_dir', '-cpd', type=str)
+args = parser.parse_args()
+ray.init(num_cpus=33, num_gpus=0,include_dashboard=True, ignore_reinit_error=True)
+
+#configures training of the agent with associated hyperparameters
+config_train = {
+            #"sample_batch_size": 200,
+            "env": Envir,
+            "train_batch_size": 1000,
+            #"sgd_minibatch_size": 1200,
+            #"num_sgd_iter": 3,
+            #"lr":1e-3,
+            #"vf_loss_coeff": 0.5,
+            #"rollout_fragment_length":  63,
+            "model":{"fcnet_hiddens": [64, 64]},
+            "num_workers": 32,
+            "env_config":{"generalize":True, "run_valid":False, "horizon":20},
+            }
+
+#Runs training and saves the result in ~/ray_results/train_ngspice_45nm
```

Comment:
results should not be saved to an absolute path, instead save them somewhere in this project, and add that directory to your `.gitignore`